### DISCUSSION TITLE
Disable automatic "re-login"

### DISCUSSION THREAD

**Discussion by arminvburren** [fief-dev/fief#49] [Q&A] [answered]
So I have integrated fief to my website https://simu-immo.fr works well (with brave 2 users right now!).

One problem is that I cant easily "change account". Why ? If I disconnect through the my logout page, and go through login again, fief will automatically connect me to the last account i entered into fief (you can probably verify this by yourself on the website right now). The only way to change account is to manually remove the fief cookie in my browser.

Is it possible to remove the fact it automatically connects on the last login ?

**Comment by frankie567** [comment, marked as answer]
Hi @arminvburren!

You need to redirect to the `/logout` endpoint of Fief to clear the session on Fief's side. You can generate with the Python client and make a 302 redirection to it after having cleared your own session: https://docs.fief.dev/integrate/python/#logout_url

Assuming FastAPI, something like this:

```py
@app.get("/logout", name="logout")
async def logout(request: Request, response: Response):
    redirect_uri = request.url_for("index")
    logout_url = fief.logout_url(redirect_uri)

    response = RedirectResponse(logout_url)  # Response redirecting to Fief logout URL
    response.delete_cookie(SESSION_COOKIE_NAME)  # Don't forget to clear our own session cookie

    return response
```

Fief will automatically redirect the user back to the URL you give (e.g. your index page).

**Reply by arminvburren** [reply]
Mm this is already what I do in the my code. Here's explicityl my logout function: 
```
@authrouter.get("/logout", name="logout")
async def logout(request: Request):
    logout_redirect = "http://localhost:8000/" if request.client.host == "127.0.0.1" else "https://simu-immo.fr/"
    logout_url = await fief.logout_url(logout_redirect)
    response = RedirectResponse(logout_url)
    response.delete_cookie("user_session_cookie")
    return response
```

**Reply by arminvburren** [reply]
It seem to me the logout url of fief isn't cleaning its own cookie.. is it ?

**Reply by arminvburren** [reply]
ok but the behavior is not appearing anymore and I always get on the fief connect page as it should be... did you change anything in the meantime on your side ?

**Reply by frankie567** [reply]
Nice to hear that you sorted it out! I didn't change anything on my side, the last deployment dates from 16 days ago.

The Fief session cookie is named `fief_session` and is tied to your subdomain, i.e. `simu-immo.fief.dev`; you can check if it's correctly cleared.
